src/ledgerwriter/src/test/java/anthos/samples/financedemo/ledgerwriter/TransactionValidatorTest.java

```diff
@@ -0,0 +1,206 @@
+                            LOCAL_ROUTING_NUM, AUTHED_ACCOUNT_NUM, transaction);
+                });
+
+        // Then
+        assertNotNull(exceptionThrown);
+        assertEquals(transactionValidator.EXCEPTION_MESSAGE_INVALID_NUMBER,
+                exceptionThrown.getMessage());
+    }
+
+    @Test
+    @DisplayName("Given invalid receiver routing number, " +
+            "IllegalArgumentException is thrown")
+    void validateTransactionFailWhenInvalidToRoutingNumber() {
+        // Given
+        when(transaction.getToRoutingNum()).thenReturn(INVALID_NUM);
+
+        // When
+        IllegalArgumentException exceptionThrown = assertThrows(
+                IllegalArgumentException.class, () -> {
+                    transactionValidator.validateTransaction(
+                            LOCAL_ROUTING_NUM, AUTHED_ACCOUNT_NUM, transaction);
+                });
+
+        // Then
+        assertNotNull(exceptionThrown);
+        assertEquals(transactionValidator.EXCEPTION_MESSAGE_INVALID_NUMBER,
+                exceptionThrown.getMessage());
+    }
+
+    @Test
+    @DisplayName("Given the sender is not authenticated, " +
+            "IllegalArgumentException is thrown")
+    void validateTransactionFailWhenNotAuthenticated() {
+        // Given
+        when(transaction.getFromAccountNum()).thenReturn(
+                NON_AUTHED_ACCOUNT_NUM);
+
+        // When
+        IllegalArgumentException exceptionThrown = assertThrows(
+                IllegalArgumentException.class, () -> {
+                    transactionValidator.validateTransaction(
+                            LOCAL_ROUTING_NUM, AUTHED_ACCOUNT_NUM, transaction);
+                });
+
+        // Then
+        assertNotNull(exceptionThrown);
+        assertEquals(transactionValidator.EXCEPTION_MESSAGE_NOT_AUTHENTICATED,
+                exceptionThrown.getMessage());
+    }
+
+    @Test
+    @DisplayName("Given the sender is the receiver, " +
+            "IllegalArgumentException is thrown")
+    void validateTransactionFailWhenSenderIsReceiver() {
+        // Given
+        when(transaction.getToAccountNum()).thenReturn(AUTHED_ACCOUNT_NUM);
+        when(transaction.getToRoutingNum()).thenReturn(LOCAL_ROUTING_NUM);
+
+        // When
+        IllegalArgumentException exceptionThrown = assertThrows(
+                IllegalArgumentException.class, () -> {
+                    transactionValidator.validateTransaction(
+                            LOCAL_ROUTING_NUM, AUTHED_ACCOUNT_NUM, transaction);
+                });
+
+        // Then
+        assertNotNull(exceptionThrown);
+        assertEquals(transactionValidator.EXCEPTION_MESSAGE_SEND_TO_SELF,
+                exceptionThrown.getMessage());
+    }
+
+    @Test
+    @DisplayName("Given an invalid transaction amount, " +
+            "IllegalArgumentException is thrown")
+    void validateTransactionFailWhenAmountNotValid() {
+        // Given
+        when(transaction.getAmount()).thenReturn(INVALIDAMOUNT);
```

Comment:
Same here, is it possible to test multiple amounts? It might be useful to test some values right around 0, 0 itself, and also very large and very small

Reply:
Added tests as well